app/models/concerns/assign_to_current_semester.rb

```diff
@@ -6,6 +6,7 @@ module AssignToCurrentSemester
   end
 
   def assign_to_current_semester
-    self.semester = Semester.current_semester unless semester.present?
+    # TODO: Settings are not initilized yet here
+    # self.semester = Semester.find(Settings.new.current_semester_id)
```

Comment:
Yeah it seems that way. But, a lot of parts in the app don't really make sense if they aren't in the context of a semester either. Maybe we can put checks for both a semester and a setting object in an initializer?